### PR TITLE
[Tensor] initialize the ColoOptimizer

### PR DESCRIPTION
**Pull request by feifeibear**

### CODE REVIEW

File: colossalai/tensor/colo_tensor.py [hunk 1]
```diff
@@ -103,7 +104,7 @@ def init_from_torch_tensor(tensor: torch.Tensor, save_payload=True, is_model_dat
                             requires_grad=tensor.requires_grad,
                             pin_memory=tensor.is_pinned(),
                             device=tensor.device,
-                            torch_tensor=tensor if save_payload else torch.empty(0),
+                            torch_tensor=tensor.detach() if save_payload else torch.empty(0),
```
Comment (Wesley-Jzy) [inline colossalai/tensor/colo_tensor.py:107]: why add detach() here?

Comment (feifeibear) [reply colossalai/tensor/colo_tensor.py:107]: optimizer need all parameters as a leaf node in the graph.
without detach() will not pass the ColoTensor to an optimizer

File: tests/test_tensor/test_model.py [hunk 2]
```diff
@@ -127,6 +129,33 @@ def __init__(self):
...
+def test_colo_optimizer():
+    get_components_func = non_distributed_component_funcs.get_callable('simple_net')
+    model_builder, train_dataloader, test_dataloader, optimizer_class, criterion = get_components_func()
+    set_seed(1)
+    with ColoInitContext(lazy_memory_allocate=False, device=get_current_device()):
+        model = model_builder(checkpoint=True)
+
+    colo_optimizer = ColoOptimizer(dict(model.named_parameters()), torch.optim.SGD, lr=0.1)
+    for i, (data, label) in enumerate(train_dataloader):
+        data = data.to(get_current_device())
```
Comment (Wesley-Jzy) [inline tests/test_tensor/test_model.py:141]: Maybe call & test zero_grad() first

Comment (feifeibear) [reply tests/test_tensor/test_model.py:141]: Fixed it!

File: colossalai/tensor/colo_tensor.py (hunk 1 above)
Comment (feifeibear) [reply colossalai/tensor/colo_tensor.py:107]: Polish it in a better way.

**Review by Wesley-Jzy — APPROVED** [review]
